mne/io/egi/egi.py

```diff
@@ -65,41 +65,24 @@ def my_fread(*x, **y):
         info['event_codes'] = np.array(info['event_codes'])
     else:
         raise NotImplementedError('Only continous files are supported')
-
     info.update(dict(precision=precision, unsegmented=unsegmented))
-
+    info['dtype'], info['bytesize'] = \
+        {2: ('>i2', 2), 4: ('>f4', 4),
+         6: ('>f8', 8)}[info['precision']]
     return info
 
 
 def _read_events(fid, info):
     """Read events"""
-    unpack = [info[k] for k in ['n_events', 'n_segments', 'n_channels']]
-    n_events, n_segments, n_channels = unpack
-    n_samples = 1 if info['unsegmented'] else info['n_samples']
-    events = np.zeros([n_events, n_segments * info['n_samples']])
-    dtype, bytesize = {2: ('>i2', 2), 4: ('>f4', 4),
-                       6: ('>f8', 8)}[info['precision']]
-
-    info.update({'dtype': dtype, 'bytesize': bytesize})
-    beg_dat = fid.tell()
-
-    for ii in range(info['n_events']):
-        fid.seek(beg_dat + (int(n_channels) + ii) * bytesize, 0)
-        events[ii] = np.fromfile(fid, dtype, n_samples)
-        fid.seek(int((n_channels + n_events) * bytesize), 1)
-    return events
```

Comment:
It was wrong because it assumed the data was stored row-major (if you think of rows as being channels and cols as being time), i.e. with one channel, then the next, then the next. But looking at the old `_read_data` function, which reads data properly, the data are stored column-major, i.e. all the channels for the first time sample, then all the channels for the next time sample, etc.